src/services/organizeImports.ts

```diff
@@ -397,14 +397,21 @@ namespace ts.OrganizeImports {
             factory.updateImportClause(importDeclaration.importClause!, importDeclaration.importClause!.isTypeOnly, name, namedBindings), // TODO: GH#18217
             importDeclaration.moduleSpecifier);
     }
-
+    
     function sortSpecifiers<T extends ImportOrExportSpecifier>(specifiers: readonly T[]) {
         return stableSort(specifiers, compareImportOrExportSpecifiers);
     }
 
     export function compareImportOrExportSpecifiers<T extends ImportOrExportSpecifier>(s1: T, s2: T) {
-        return compareIdentifiers(s1.propertyName || s1.name, s2.propertyName || s2.name)
+        const comparison = compareIdentifiers(s1.propertyName || s1.name, s2.propertyName || s2.name)
             || compareIdentifiers(s1.name, s2.name);
+
+        if (comparison === Comparison.LessThan) {
+            // Ignore new lines if the node is going to get rearranged.
+            ignoreSourceNewlines(s1);

```

Comment:
I don't like a side effect in a sort comparator, especially one that's one-sided like this (imagine if the sort algorithm changes and we happen to only get GreaterThan results?). Can we move this to a post-pass once the list is sorted, or is there some other invariant that makes this robust against how the sort algorithm might be implemented?

Reply:
I have forced pushed a new solution to the issue. This code is no longer valid.